### PR TITLE
[KAFKA] Update README.md based on template

### PR DESCRIPTION
**Pull request by mbdcos**
This is the initial update for new README.md (new Kafka service guide based on new template). Please change as necessary.

 README.md template mentions  ‘node’. This can be confusing for Kafka. In Kafka, we have broker tasks only. You can run multiple brokers on the same Mesos node.

### CODE REVIEW

File: frameworks/kafka/README.md [hunk 1]
```diff
@@ -1,14 +1,420 @@
...
+# Support
+
+## Supported Versions
+
+- Apache Kafka: _WHAT VERSION OF YOUR SERVICE IS INCLUDED IN THE PACKAGE?_
+- DC/OS: _LIST VERSION(S) OF DC/OS THAT YOU'VE TESTED AND SUPPORT_
+
+## Package Versioning
+
+Packages are versioned with an `a.b.c-x.y.z` format, where `a.b.c` is the version of the service management layer and `x.y.z` indicates the version of Apache Kafka. For example, `1.5.0-3.2.1` indicates version `1.5.0` of the service management layer and version `3.2.1` of Apache Kafka.
```
Comment (triclambert) [inline frameworks/kafka/README.md]: Should use real versions here, for example: `1.1.19.1-0.10.1.0`.

File: frameworks/kafka/README.md [hunk 2]
```diff
@@ -1,14 +1,420 @@
...
+
+### Upgrades/downgrades
+
+The package supports upgrade and rollback between adjacent versions only. For example, to upgrade from version 2 to version 4, you must first complete an upgrade to version 3, followed by an upgrade to version 4.
```
Comment (triclambert) [inline frameworks/kafka/README.md]: Link to the upgrade doc.

**Comment by triclambert** [comment]
General comment:  please update all of the generic text which is identified with underscores.

File: frameworks/kafka/README.md [hunk 3]
```diff
@@ -1,14 +1,420 @@
...
+- [Support](#support)
+  - Supported Versions
+  - Package Versioning
+    - Upgrades/downgrades
+  - Reaching Technical Support
+
+<a name="overview"></a>
+# Overview
+
+DC/OS Apache Kafka is an automated service that makes it easy to deploy and manage Apache Kafka on [Mesosphere DC/OS](http://dcos.io). Apache Kafka is a distributed high-throughput publish-subscribe messaging system with strong ordering guarantees. Kafka clusters are highly available, fault tolerant, and very durable. For more information on Apache Kafka, see its [documentation](http://kafka.apache.org/documentation.html).
```
Comment (triclambert) [inline frameworks/kafka/README.md]: @sascala isn't the correct name "Enterprise DC/OS"?

File: frameworks/kafka/README.md [hunk 4]
```diff
@@ -1,14 +1,420 @@
...
+The service comes with a reasonable initial configuration for evaluation use. Additional customizations may be made to the service configuration at initial install, and later updated once the service is already running through a configuration rollout process. If you just want to try out the service, you can use the default configuration and be up and running within moments.
+
+Interoperating clients and services may take advantage of DC/OS service discovery features to directly access Apache Kafka via advertised endpoints, regardless of where the instance is currently located within a DC/OS Cluster.
+
+Multiple instances can be installed on DC/OS and managed independently. This allows different teams within an organization to have isolated instances of the service.
+
+## Features
+
+- Multiple instances sharing the same physical systems (requires custom port configuration).
+- Vertical (resource) and horizontal (node count) scaling.
```
Comment (triclambert) [inline frameworks/kafka/README.md]: Perhaps "adding node count"?

File: frameworks/kafka/README.md [hunk 5]
```diff
@@ -1,14 +1,420 @@
...
+
+Each port exposed by the service may be customized via the service configuratiton. If you wish to install multiple instances of the service and have them colocate on the same machines, you must ensure that **no** ports are common between those instances. Customizing ports is only needed if you require multiple instances sharing a single machine. This customization is optional otherwise.
+
+### Storage Volumes
+
+The service supports two volume types:
+- `ROOT` volumes are effectively an isolated directory on the root volume, sharing IO/spindles with the rest of the host system.
+- `MOUNT` volumes are a dedicated device or partition on a separate volume, with dedicated IO/spindles.
+
+Using `MOUNT` volumes requires [additional configuration on each DC/OS agent system](https://dcos.io/docs/1.8/administration/storage/mount-disk-resources/), so the service currently uses `ROOT` volumes by default. To ensure reliable and consistent performance in a production environment, you should configure `MOUNT` volumes on the machines which will run the service in your cluster and then configure the following as `MOUNT` volumes:
```
Comment (triclambert) [inline frameworks/kafka/README.md]: The last sentence here is incomplete, ending in a colon.

File: frameworks/kafka/README.md [hunk 6]
```diff
@@ -1,14 +1,420 @@
...
+
+	```
+	dcos kafka pods replace _NODEPOD_-3
+	```
+
+<a name="disaster-recovery"></a>
+# Disaster Recovery
+
+## Backup/Restore
+  For more information on Apache Kafka backup/restore, please see its [documentation](http://kafka.apache.org/documentation.html)
```
Comment (triclambert) [inline frameworks/kafka/README.md]: Maybe add something like "The DC/OS Apache Kafka does not natively support any backup or restore functionality. Backup and restore are critical to any disaster recovery plan."

File: frameworks/kafka/README.md [hunk 7]
```diff
@@ -1,14 +1,420 @@
...
+  
+<a name="#deploy-best-practices"></a>
+# Deployment Best Practices
+
+- Run [backups](#disaster-recovery) on a regular basis, and test your backups.
+- Configure alerting/monitoring of your service to detect downtime and other issues.
+- If your cluster has been [configured with availability zones (e.g. Rack IDs)](https://github.com/dcos/dcos-docs/blob/51fe4641152e2c9361877439c40ddfeab61506e0/1.8/administration/faq.md#q-how-to-add-mesos-attributes-to-nodes-in-order-to-use-marathon-constraints), Placement Constraints may be used to map the service across those zones.
```
Comment (triclambert) [inline frameworks/kafka/README.md]: I think we should remove this.  @keithchambers

File: frameworks/kafka/README.md [hunk 8]
```diff
@@ -1,14 +1,420 @@
...
+1. Click the `Frameworks` tab in the upper left to get a list of services running in the cluster.
+1. Navigate into the correct Framework for your needs. The Scheduler runs under `marathon` with a task name matching the service name (default _`PKGNAME`_). Meanwhile service nodes run under a Framework whose name matches the service name (default _`PKGNAME`_).
+1. You should now see two lists of tasks. `Active Tasks` are what's currently running, and `Completed Tasks` are what has since exited. Click on the `Sandbox` link for the task you wish to examine.
+1. The `Sandbox` view will list files named `stdout` and `stderr`. Click the file names to view the files in the browser, or click `Download` to download them to your system for local examination. Note that very old tasks will have their Sandbox automatically deleted to limit disk space usage.
+
+
+<a name="limitations"></a>
+# Limitations
+
+- Shrinking cluster size (number of brokers) is not supported.
```
Comment (triclambert) [inline frameworks/kafka/README.md]: This seems to be a dupe of the section immediately following it.

File: frameworks/kafka/README.md [hunk 9]
```diff
@@ -1,14 +1,420 @@
...
+
+_FOR EXAMPLE, DOES YOUR SERVICE REQUIRE MANUAL INVOLVEMENT BY THE USER IN CERTAIN SCENARIOS?_
+
+<a name="terms"></a>
+# Terms of Use
+
+_ANY RISK WARNINGS OR REQUIREMENTS FOR SUPPORTED ENVIRONMENTS GO HERE_
+
+<a name="support"></a>
+# Support
```
Comment (triclambert) [inline frameworks/kafka/README.md]: "Enterprise DC/OS customers may submit support cases via support@mesosphere.com."

File: frameworks/kafka/README.md [hunk 10]
```diff
@@ -1,14 +1,420 @@
...
-- [Limitations](docs/limitations.md)
+- [Overview](#overview)
+  - Features
+- [Quick Start](#quick-start)
+- [Installing and Customizing](#installing-and-customizing)
+  - Installation with Default Settings
+  - Installation with Custom Settings
+  - Service Settings
+    - Service Name
+    - _SERVICE-WIDE OPTIONS SPECIFIC TO YOUR IMPLEMENTATION GO HERE_
```
Comment (nickbp) [inline frameworks/kafka/README.md]: Should update the TOC to reflect the content below. For example some of these suggested sections can be cleaned up, and 'node settings' was renamed to 'broker settings'

File: frameworks/kafka/README.md (hunk 3 above)
Comment (sascala) [inline frameworks/kafka/README.md]: Yes, and the link should got to our enterprise site, not the OSS site. I will fix.

File: frameworks/kafka/README.md (hunk 6 above)
Comment (mbdcos) [inline frameworks/kafka/README.md]: "does not support native backup/restore"  we should not say such a thing I guess

File: frameworks/kafka/README.md (hunk 2 above)
Comment (mbdcos) [inline frameworks/kafka/README.md]: Upgrade doc became release notes.

File: frameworks/kafka/README.md (hunk 1 above)
Comment (mbdcos) [inline frameworks/kafka/README.md]: 1.1.20-0.10.1.0

File: frameworks/kafka/README.md (hunk 9 above)
Comment (mbdcos) [inline frameworks/kafka/README.md]: ok

File: frameworks/kafka/README.md (hunk 8 above)
Comment (mbdcos) [inline frameworks/kafka/README.md]: removed

File: frameworks/kafka/README.md (hunk 7 above)
Comment (mbdcos) [inline frameworks/kafka/README.md]: done

File: frameworks/kafka/README.md (hunk 5 above)
Comment (mbdcos) [inline frameworks/kafka/README.md]: removed

File: frameworks/kafka/README.md (hunk 4 above)
Comment (mbdcos) [inline frameworks/kafka/README.md]: broker count

File: frameworks/kafka/README.md (hunk 6 above)
Comment (triclambert) [inline frameworks/kafka/README.md]: Please add "The DC/OS Apache Kafka does not natively support any backup or restore functionality."  I'd like that there explicitly.

Comment (mbdcos) [inline frameworks/kafka/README.md]: added

**Comment by mbdcos** [comment]
retest this please

File: frameworks/kafka/README.md [hunk 11]
```diff
@@ -1,14 +1,350 @@
...
-- [Managing](docs/managing.md)
-- [API Reference](docs/api-reference.md)
-- [Troubleshooting](docs/troubleshooting.md)
-- [Limitations](docs/limitations.md)
+<p align="left"><img src="https://mesosphere.com/wp-content/themes/mesosphere/library/images/assets/dcos-sdk-logo.png" width="250"/></p>
+
+
+# Apache Kafka Service Guide
+
+<!-- Add TOC later if needed, once all sections are finalized
```
Comment (sascala) [inline frameworks/kafka/README.md]: This comment wasn't terminated, making the whole doc a comment. I terminated in my edit. Though I'll need someone to read through to make sure I haven't uncommented things that should be commented.

File: frameworks/kafka/README.md [hunk 12]
```diff
@@ -1,14 +1,350 @@
...
+1. Navigate to `Universe` => `Packages` and find the `kafka` package.
+1. Click `Install`, then in the pop up dialog click `Advanced` to see the customization dialog.
+1. Make your changes to the default configuration in the customization dialog, then click `Review`.
+1. Examine the configuration summary for any needed changes. Click `Back` to make changes, or `Install` to confirm the settings and install the service.
+
+## Service Settings
+
+### Service Name
+
+Each instance of Apache Kafka in a given DC/OS cluster must be configured with a different service name. You can configure the service name in the **service** section of the install settings. The default service name (used in many examples here) is `kafka`.
```
Comment (sascala) [inline frameworks/kafka/README.md]: Do these locations, e.g., "the service section," refer to the UI?

File: frameworks/kafka/README.md [hunk 13]
```diff
@@ -1,14 +1,350 @@
...
+The amount of CPU allocated to each broker may be customized. A value of `1.0` equates to one full CPU core on a machine. This value may be customized by editing the **cpus** value under the **broker** configuration section. Turning this too low will result in throttled tasks.
+
+### Memory
+
+The amount of RAM allocated to each broker may be customized. This value may be customized by editing the **mem** value (in MB) under the **broker** configuration section. Turning this too low will result in out of memory errors.
+
+
+### Ports
+
+Each port exposed by the service may be customized via the service configuratiton. If you wish to install multiple instances of the service and have them colocate on the same machines, you must ensure that **no** ports are common between those instances. Customizing ports is only needed if you require multiple instances sharing a single machine. This customization is optional otherwise.
```
Comment (sascala) [inline frameworks/kafka/README.md]: do we mean the service configuration section of the UI advanced installation page?

**Comment by mbdcos** [comment]
retest this please

**Comment by mbdcos** [comment]
retest this please

**Comment by mbdcos** [comment]
retest this please

**Comment by mbdcos** [comment]
CI is broken. Do not expect tests to complete/pass. This is a documentation PR, does not touch any code/script so testing is irrelevant. Should be fine to merge without waiting CI to be fixed.

**Comment by mbdcos** [comment]
To be closed on Monday if not approved (more than a week old - internal PR policy)

**Pull request merged**